lib/attributor/types/hash.rb

```diff
@@ -1,4 +1,20 @@
 module Attributor
+  class InvalidDefinition < StandardError
+    def initialize(type, cause)
+      type_name = if type.name
+        type.name
+      else
+        type.inspect
+      end
+
+      msg = "Type #{type_name} is invalid due to exception during definition: #{cause.inspect}."
```

Comment:
maybe a more refined message?
"Structure definition for type #{type_name} is invalid. The following exception has occurred: #{cause.inspect}"